apps/ios/SimpleX SE/ShareModel.swift

```diff
@@ -71,13 +77,58 @@ class ShareModel: ObservableObject {
                 return .failure(.unsupportedFormat)
             }
             let url = try await attachment.inPlaceUrl(type: type)
-            guard let cryptoFile = saveFileFromURL(url) else { return .failure(.encryptFile) }
+            var cryptoFile: CryptoFile
+            var msgContent: MsgContent
+            switch type {
+
+            // Prepare Image message
+            case .image:
+
+                // Animated
+                if attachment.hasItemConformingToTypeIdentifier(UTType.gif.identifier) {
+                    guard let data = try? Data(contentsOf: url), let image = UIImage(data: data) else {
+                        return .failure(.mediaLoading)
+                    }
+                    guard let file = saveAnimImage(image) else {
+                        return .failure(.encryptFile)
+                    }
+                    guard let imageString = resizeImageToStrSize(image, maxDataSize: MAX_DATA_SIZE) else {
+                        return .failure(.previewGeneration)
+                    }
+                    cryptoFile = file
+                    msgContent = .image(text: comment, image: imageString)
+
+                // Static
+                } else {
+                    guard let image = downsampleImage(at: url, to: MAX_DOWNSAMPLE_SIZE) else {
+                        return .failure(.imageDownsampling)
+                    }
+                    guard let file = saveImage(image) else {
+                        return .failure(.encryptFile)
+                    }
+                    guard let imageString = resizeImageToStrSize(image, maxDataSize: MAX_DATA_SIZE) else {
+                        return .failure(.previewGeneration)
+                    }
+                    cryptoFile = file
+                    msgContent = .image(text: comment, image: imageString)
+                }
+
+            // Prepare Data message
+            case .data:
+                msgContent = .file(comment)
+                guard let file = saveFileFromURL(url) else { return .failure(.encryptFile) }
```

Comment:
this is not necessarily encryption error, it can be IO error, so it should be a generic "Error saving file"